test/fakeintake/server/server.go

```diff
@@ -67,6 +69,8 @@ func NewServer(options ...func(*Server)) *Server {
 		opt(fi)
 	}
 
+	go fi.cleanUpPayloads()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Have you considered using time ticket instead of sleeping in the go routine ? It helps cleaning up properly subroutines, sending a `stop` signal to the channel

https://gobyexample.com/tickers

Reply:
It did not considered since I did not even know about it. It looks to be the best thing to use for my use case. I will make the changes. Thanks!